packages/i18n/src/create-i18n.js

```diff
@@ -186,12 +200,37 @@ export const createI18n = ( initialData, initialDomain, hooks ) => {
 		};
 	};
 
+	/**
+	 * @param {LocaleData} [data]
+	 * @param {string}     [domain]
+	 */
+	const doAddLocaleData = ( data, domain = 'default' ) => {
+		tannin.data[ domain ] = {
+			...DEFAULT_LOCALE_DATA,
+			...tannin.data[ domain ],
+			...data,
+			// Populate default domain configuration (supported locale date which omits
+			// a plural forms expression).
+			'': {
+				...DEFAULT_LOCALE_DATA[ '' ],
+				...( tannin.data[ domain ] || {} )[ '' ],
+				...( data || {} )[ '' ],
+			},
+		};
+	};
+
 	/** @type {SetLocaleData} */
 	const setLocaleData = ( data, domain ) => {
 		doSetLocaleData( data, domain );
 		notifyListeners();
 	};
 
+	/** @type {AddLocaleData} */
+	const addLocaleData = ( data, domain ) => {
+		doAddLocaleData( data, domain );
```

Comment:
The `doAddLocaleData` function could be inlined here. There is a standalone `doSetLocaleData` function because we sometimes want to do the operation without firing notifications, but `addLocaleData` doesn't have that requirement.

Reply:
Now you can remove `doAddLocaleData` completely, and do the `tannin` assignment directly in the `addLocaleData` body.